rust-runtime/aws-smithy-json/src/deserialize/token.rs

```diff
@@ -197,12 +193,7 @@ pub fn expect_number_or_null(
 pub fn expect_blob_or_null(token: Option<Result<Token<'_>, Error>>) -> Result<Option<Blob>, Error> {
     Ok(match expect_string_or_null(token)? {
         Some(value) => Some(Blob::new(base64::decode(value.as_escaped_str()).map_err(
-            |err| {
-                Error::new(
-                    ErrorReason::Custom(Cow::Owned(format!("failed to decode base64: {}", err))),
-                    None,
-                )
-            },
+            |err| Error::custom(format!("failed to decode base64: {}", err)),
```

Comment:
Should this `err` be interpolated as part of the custom message?
It feels like it should be used as a source?